speechbrain/decoders/seq2seq.py

```diff
@@ -1856,3 +1856,108 @@ def forward_step(self, inp_tokens, memory, enc_states, enc_lens):
+            Normally set to None for this usecase.
+    temperature: float
+        decoding temperature
+    temperature_lm: float
+        temperature for the model LM head.
+    bos_token: int
+        BOS token ID. 0 by default.
+    pad_token: int
+        Padding token ID. 1 by default.
+    eos_token: int
+        EOS token ID. 2 by default.
+    normalize: bool
+        Whether or not to normailze the text.
+    **kwargs
+        Arguments to pass to S2SBeamSearcher
+    """
+
+    def __init__(
+        self,
+        module,
+        temperature=1.0,
+        temperature_lm=1.0,
+        bos_token=0,
+        pad_token=1,
+        eos_token=2,
+        normalize=False,
+        **kwargs,
+    ):
+
+        super(S2SSpeechT5BeamSearch, self).__init__(**kwargs)
+
+        self.model = module[0]
+        if len(module) == 2:
+            self.ctc_fc = module[1]
+
+        self.softmax = torch.nn.LogSoftmax(dim=-1)
+
+        self.temperature = temperature
+        self.temperature_lm = temperature_lm
+
+        self.decoder_input_tokens = [bos_token]
+
+        self.bos_token = bos_token
+        self.pad_token = pad_token
+        self.eos_token = eos_token
+        self.normalize = normalize
+
+    def set_decoder_input_tokens(self, decoder_input_tokens):
+        """decoder_input_tokens are the tokens used as input to the decoder."""
+        self.set_bos_token(decoder_input_tokens[0])
+        self.decoder_input_tokens = [self.bos_token]
+
+    def reset_mem(self, batch_size, device):
+        """This method sets the first tokens to be decoder_input_tokens during search."""
+        return torch.tensor([self.decoder_input_tokens] * batch_size).to(device)
+
+    def reset_lm_mem(self, batch_size, device):
+        """Needed to reset the LM memory during beamsearch."""
+        return None
+
+    def permute_mem(self, memory, index):
+        """Permutes the memory."""
+        memory = torch.index_select(memory, dim=0, index=index)
+        return memory
+
+    def permute_lm_mem(self, memory, index):
+        """Permutes the memory of the language model."""
+        memory = torch.index_select(memory, dim=0, index=index)
+        return memory
+
+    def forward_step(self, inp_tokens, memory, enc_states, enc_lens):
+        """Performs a step in the implemented beamsearcher."""
+        memory = _update_mem(inp_tokens, memory)
+        (dec_out, attn,) = self.model.forward_decoder(enc_states, memory)
+        log_probs = self.softmax(dec_out[:, -1])
+        return log_probs, memory, attn
+
+    def lm_forward_step(self, inp_tokens, memory):
+        """Performs a step in the implemented LM module."""
+        memory = _update_mem(inp_tokens, memory)
+        if not next(self.lm_modules.parameters()).is_cuda:
+            self.lm_modules.to(inp_tokens.device)
+        logits = self.lm_modules(memory)
+        log_probs = self.softmax(logits / self.temperature_lm)
+        return log_probs[:, -1, :], memory
```

Comment:
this is not required since now you can have LM scorers

Reply:
Hello, should I just remove it or make changes to this method ?